pkgs/by-name/do/dooit/package.nix

```diff
@@ -5,53 +5,62 @@
   python311,
   testers,
   nix-update-script,
+  extraPackages ? [ ],
 }:
 let
   python3 = python311;
+  username = "dooit-org";
+  repo = "dooit";
+  ver = "v3.0.2";
 in
 python3.pkgs.buildPythonApplication rec {
-  pname = "dooit";
-  version = "2.2.0";
+  pname = repo;
+  version = ver;
   pyproject = true;
 
   src = fetchFromGitHub {
-    owner = "kraanzu";
-    repo = "dooit";
-    rev = "v${version}";
-    hash = "sha256-GtXRzj+o+FClleh73kqelk0JrSyafZhf847lX1BiS9k=";
+    owner = username;
+    repo = pname;
+    rev = ver;
+    hash = "sha256-DPmCADFduGc5n+6q9zl0f4x9C6RmzLvBeYh2j0ZSpH0=";
   };
 
   build-system = with python3.pkgs; [ poetry-core ];
 
   pythonRelaxDeps = [
     "tzlocal"
+    "textual"
+    "sqlalchemy"
   ];
 
-  propagatedBuildInputs = with python3.pkgs; [
-    appdirs
-    pyperclip
-    python-dateutil
-    pyyaml
-    (textual.overridePythonAttrs (oldAttrs: {
-      version = "0.47.1";
-      src = fetchFromGitHub {
-        owner = "Textualize";
-        repo = "textual";
-        rev = "refs/tags/v0.47.1";
-        hash = "sha256-RFaZKQ+0o6ZvfZxx95a1FjSHVJ0VOIAfzkdxYQXYBKU=";
-      };
-      disabledTests = [
-        "test_tracked_slugs"
-        "test_textual_env_var"
-        "test_register_language"
-        "test_register_language_existing_language"
-      ];
-    }))
-    tzlocal
-  ];
+  propagatedBuildInputs =
+    with python3.pkgs;
+    [
+      pyperclip
+      textual
+      pyyaml
+      dateutil
+      sqlalchemy
+      platformdirs
+      tzlocal
+      click
+    ]
+    ++ extraPackages;
+
+  # testing
 
-  # No tests available
-  doCheck = false;
+  preBuild = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  nativeCheckInputs = with python3.pkgs; [
+    pytest
+    faker
+  ];
+  doCheck = true;
+  checkPhase = ''
+    python -m pytest
+  '';
```

Comment:
Do we need all this or could we get away with `pytestCheckHook`?

Reply:
Didn't know about this. I'll check it out